apps/condo/domains/banking/schema/BankAccountReportTask.test.js

```diff
@@ -0,0 +1,300 @@
+const { createTestOrganization } = require('@condo/domains/organization/utils/testSchema')
+const { createTestOrganizationEmployeeRole, createTestOrganizationEmployee } = require('@condo/domains/organization/utils/testSchema')
+const { makeClientWithNewRegisteredAndLoggedInUser, makeClientWithSupportUser } = require('@condo/domains/user/utils/testSchema')
+
+const { BANK_INTEGRATION_IDS } = require('../constants')
+const { createTestBankIntegrationAccountContext, createTestBankContractorAccount, BankIntegration, createTestBankCategory, createTestBankCostItem } = require('../utils/testSchema')
+
+
+function generateDateArray (dateStart, interval) {
+    const dateInterval = [dateStart]
+    for (let i = 1; i <= interval; i++) {
+        dateInterval.push(dayjs(dateInterval[dateInterval.length - 1]).add(1, 'day').format('YYYY-MM-DD'))
+    }
+    return dateInterval
+}
+
+describe('BankAccountReportTask', () => {
+    let admin, bankIntegration, category
+    beforeAll(async () => {
+        admin = await makeLoggedInAdminClient()
+        bankIntegration = await BankIntegration.getOne(admin, { id: BANK_INTEGRATION_IDS['1CClientBankExchange'] });
+        ([category] = await createTestBankCategory(admin))
+
+    })
+    describe('CRUD tests', () => {
+        describe('create', () => {
+            test('admin can', async () => {
+                // 1) prepare data
+                const [org] = await createTestOrganization(admin)
+                const [account] = await createTestBankAccount(admin, org)
+                // 2) action
+                const [obj, attrs] = await createTestBankAccountReportTask(admin, account, org, admin.user.id, { progress: 0 })
+
+                // 3) check
+                expect(obj.id).toMatch(UUID_RE)
+                expect(obj.dv).toEqual(1)
+                expect(obj.sender).toEqual(attrs.sender)
+                expect(obj.v).toEqual(1)
+                expect(obj.newId).toEqual(null)
+                expect(obj.deletedAt).toEqual(null)
+                expect(obj.createdBy).toEqual(expect.objectContaining({ id: admin.user.id }))
+                expect(obj.updatedBy).toEqual(expect.objectContaining({ id: admin.user.id }))
+                expect(obj.createdAt).toMatch(DATETIME_RE)
+                expect(obj.updatedAt).toMatch(DATETIME_RE)
+                expect(obj.account.id).toEqual(account.id)
+                expect(obj.status).toEqual(BANK_SYNC_TASK_STATUS.PROCESSING)
+                expect(obj.organization.id).toEqual(org.id)
+            })
+
+            test('support cannot create for another user', async () => {
+                const client = await makeClientWithSupportUser()
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await createTestBankAccountReportTask(client, { id: faker.datatype.uuid() }, { id: faker.datatype.uuid() }, faker.datatype.uuid())
+                })
+            })
+
+            test('user can', async () => {
```

Comment:
Add another tests:
- user can if it is an employee of organization with "canManageBankAccounts" permission
- user cannot if it is an employee of organization without "canManageBankAccounts" permission
- user cannot if it is an employee of another organization with "canManageBankAccounts" permission
- user can if it is an employee of linked organization with "canManageBankAccounts" permission
- user cannot if it is an employee of linked organization without "canManageBankAccounts" permission

Just copy them from BankSyncTask.test.js

Reply:
- user can if it is an employee of organization with "canManageBankAccounts" permission
I see no reason to write such tests. If the logic restricting changes to the canManage permission doesn't work, then the test that checks for it won't fail. It makes sense to write tests that, in the absence of this access, expect a fall. Then if this logic breaks and the user has access without permission, this test will fail and we will pay attention to it.

The rest I think will be useful for controlling the operation of this feature.

Reply:
It's a good practice to test features as a black box.
To test it as fully as possible, a good practice is to decompose a set of input data into equivalence classes.
For example, when we are testing ability to mange something according to some ability, then we have 2 classes of equivalence of input data:
1. Ability is presented
2. Ability is not presented 

If in tests you are making an assumptions about the code, — it will be soon or later a case, when tests will return false negative (fail, when should not) and false positive (pass when should fail) results. And these results can not be dependent from each other.

When you have no time, then ok.